### PR TITLE
doc: fix broken doc links

### PR DESCRIPTION
**Pull request by poliorcetics**
Some broken doc links I saw when compiling with `cargo +stable doc --no-deps --document-private-items --workspace --open`

### CODE REVIEW

File: crates/nu-plugin-engine/src/process.rs [hunk 1]
```diff
@@ -79,7 +79,7 @@ impl PluginProcess {
         }
     }
 
-    /// Move the plugin process out of the foreground. See [`ForegroundGuard::reset`].
```
Comment (sholderbach) [inline crates/nu-plugin-engine/src/process.rs:82]: Does this member still exist? Maybe it has the more specific doccomment itself.

File: crates/nu-table/src/table.rs [hunk 2]
```diff
@@ -847,7 +847,7 @@ fn truncate_columns_by_head(
     widths
 }
 
-/// The same as [`tabled::peaker::PriorityMax`] but prioritizes left columns first in case of equal width.
```
Comment (sholderbach) [inline crates/nu-table/src/table.rs:850]: Same, if this type still exists in the [`tabled` crate](https://docs.rs/tabled) maybe worth still mentioning it as a literal text without link.

**Review by sholderbach — COMMENTED** [review]
Thanks for cleaning this up.

File: crates/nu-plugin-engine/src/process.rs (hunk 1 above)
Comment (poliorcetics) [reply crates/nu-plugin-engine/src/process.rs:82]: Doesn't exist from what I see. There is `reset_internal` though, but it only says it must be called only once within `drop`

File: crates/nu-table/src/table.rs (hunk 2 above)
Comment (poliorcetics) [reply crates/nu-table/src/table.rs:850]: I thought `tabled` was a module, not the crate, my bad, fixed

File: crates/nu-plugin-engine/src/process.rs (hunk 1 above)
Comment (sholderbach) [reply crates/nu-plugin-engine/src/process.rs:82]: OK, then it's probably fine to just point to the `ForegroundGuard` around the meaning of foreground.

Comment (poliorcetics) [reply crates/nu-plugin-engine/src/process.rs:82]: done

**Pull request merged**